src/server/raftkv2/mod.rs

```diff
@@ -195,9 +195,21 @@ impl<EK: KvEngine, ER: RaftEngine> tikv_kv::Engine for RaftKv2<EK, ER> {
         let mut cmd = RaftCmdRequest::default();
         cmd.set_header(header);
         cmd.set_requests(vec![req].into());
-        let f = self.router.snapshot(cmd);
+        let res: tikv_kv::Result<()> = (|| {
```

Comment:
why not return err directly when enable this failpoint.

Reply:
because this function should return a future. So we can't just return another aync block eigher as its types is different from the other async block.